packages/pricing/src/models/price-set-money-amount.ts

```diff
@@ -45,12 +46,12 @@ export default class PriceSetMoneyAmount {
   money_amount: MoneyAmount
 
   @Property({ columnType: "integer", default: 0 })
-  rules_count: number
+  rules_count: number = 0
 
   @OneToMany({
     entity: () => PriceRule,
     mappedBy: (pr) => pr.price_set_money_amount,
-    cascade: ["soft-remove"] as any,
+    cascade: [Cascade.REMOVE, "soft-remove" as Cascade],
```

Comment:
comment: We have to be careful with Cascade.REMOVE, it is an application level cascade, that means that if the relation is not loaded when performing the delete, it will not be cascaded. I suggest to also look for the different configuration to apply on the different relation for database level cascade such as onDelete set null or onDelete cascade etc which will then be applied at the db level in the migration.

cc @olivermrbl